arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt

```diff
@@ -1521,3 +1521,17 @@ public inline fun <E, reified T : Throwable, A> Either<Throwable, A>.catch(@Buil
   contract { callsInPlace(catch, InvocationKind.AT_MOST_ONCE) }
   return recover { e -> if (e is T) catch(e) else throw e }
 }
+
```

Comment:
nit: copy the contracts from `leftOrNull` and `getOrNull`

Reply:
Contracts are not allowed for operator functions

Reply:
Oh actually they fixed this issue after years